module-4/metascan/README.md

```diff
@@ -0,0 +1,94 @@
+- **Command Line Interface**: Supports command-line arguments for specifying the website URL (`-u`) and output CSV file name (`-n`).
+
+## Function Descriptions
+
+### `extract_pdf_links(url)`
+
+This function takes a URL as an input and extracts all PDF links from the webpage. It uses the `requests` module to fetch the page content and `BeautifulSoup` to parse the HTML.
+
+- **Parameters**:
+  - `url` (str): The URL of the webpage to scan for PDF files.
+- **Returns**:
+  - `pdf_links` (list): A list of URLs to PDF files found on the webpage.
+
+### `download_pdf(url, save_path)`
+
+This function downloads a PDF file from the provided URL and saves it to the specified path.
+
+- **Parameters**:
+  - `url` (str): The URL of the PDF file to download.
+  - `save_path` (str): The local file path to save the downloaded PDF.
+- **Outputs**:
+  - The function prints a confirmation message indicating the download is complete.
+
+### `extract_pdf_metadata(file_path)`
+
+This function extracts metadata from a PDF file. It reads the PDF file using `PyPDF2` and retrieves various metadata attributes.
+
+- **Parameters**:
+  - `file_path` (str): The local file path of the PDF file to extract metadata from.
+- **Returns**:
+  - `metadata` (dict): A dictionary containing metadata fields such as title, author, creation date, etc.
+
+### `main()`
+
+This is the main function that orchestrates the entire process. It parses command-line arguments, extracts PDF links from the given URL, downloads each PDF, extracts metadata, and saves the results into a CSV file.
+
+- **Command-Line Arguments**:
+  - `-u` / `--url` (str): The URL of the website to scan for PDF files.
+  - `-n` / `--name` (str): The name and path of the output CSV file.
+- **Outputs**:
+  - The function prints messages to indicate progress, such as downloaded files and saved metadata.
+
+## Usage
+
+1. **Setup Environment**:
+    - Ensure Python 3.x and necessary libraries (`requests`, `beautifulsoup4`, `PyPDF2`) are installed.
+
+2. **Run the Script**:
+
+    ```bash
+    python3 metascan.py -u <website_url> -n <output_csv_filename>
+    ```
+
+    - Replace `<website_url>` with the URL of the website containing PDF files.
+    - Replace `<output_csv_filename>` with the desired name and path for the output CSV file.
+
+3. **Output**:
+    - The script will create a directory `pdfs/` to store downloaded PDF files.
+    - It will generate a CSV file containing metadata for each PDF file found, with each metadata attribute displayed on a new line for readability.
+
+## Example
+
+Suppose you want to extract PDF metadata from `https://example.com` and save it to `example_metadata.csv`:
+
+```bash
+python3 metascan.py -u https://example.com -n example_metadata.csv
+```
+
+## Dependencies
+
+- `requests`: For making HTTP requests to fetch web pages.
+- `beautifulsoup4`: For parsing HTML and extracting data from web pages.
+- `PyPDF2`: For extracting metadata from PDF files.
```

Comment:
should be listed under prerequisites, or alternatively link from prerequisites to this section